pkg/otbuiltin/init.go

```diff
@@ -76,16 +72,16 @@ func Init(path string, options initOptions) (bool, error) {
 	return true, nil
 }
 
-// Converts the mode string to a C.OSTREE_REPO_MODE enum value
-func parseMode() error {
-	if strings.EqualFold(initOpts.Mode, "bare") {
-		initOpts.repoMode = C.OSTREE_REPO_MODE_BARE
-	} else if strings.EqualFold(initOpts.Mode, "bare-user") {
-		initOpts.repoMode = C.OSTREE_REPO_MODE_BARE_USER
-	} else if strings.EqualFold(initOpts.Mode, "archive-z2") {
-		initOpts.repoMode = C.OSTREE_REPO_MODE_ARCHIVE_Z2
-	} else {
-		return errors.New("Invalid option for mode")
+// parseMode converts a mode string to a C.OSTREE_REPO_MODE enum value
+func parseMode(modeLabel string) (C.OstreeRepoMode, error) {
+	if strings.EqualFold(modeLabel, "bare") {
+		return C.OSTREE_REPO_MODE_BARE, nil
+	} else if strings.EqualFold(modeLabel, "bare-user") {
+		return C.OSTREE_REPO_MODE_BARE_USER, nil
+	} else if strings.EqualFold(modeLabel, "archive-z2") {
+		return C.OSTREE_REPO_MODE_ARCHIVE_Z2, nil
 	}
```

Comment:
There is also now `ostree_repo_mode_from_string()`.

Reply:
Ah, thanks, I can switch to that.
According to symbol versioning that appeared in 2016.3; do we aim at some specific version compatibility here in this Go library?

Reply:
If we want to be conservative, probably the [Debian stable version](https://packages.debian.org/stretch/ostree) is an OK baseline, though that is going to get increasingly painful.